pkg/spoke/managedcluster/joining_controller.go

```diff
@@ -128,25 +128,32 @@ func (c *managedClusterJoiningController) getClusterResources() (capacity, alloc
 		return nil, nil, err
 	}
 
-	cpuCapacity := *resource.NewQuantity(int64(0), resource.DecimalSI)
-	memoryCapacity := *resource.NewQuantity(int64(0), resource.BinarySI)
-	cpuAllocatable := *resource.NewQuantity(int64(0), resource.DecimalSI)
-	memoryAllocatable := *resource.NewQuantity(int64(0), resource.BinarySI)
+	capacityList := make(map[clusterv1.ResourceName]resource.Quantity)
+	allocatableList := make(map[clusterv1.ResourceName]resource.Quantity)
+
 	for _, node := range nodes {
-		cpuCapacity.Add(*node.Status.Capacity.Cpu())
-		memoryCapacity.Add(*node.Status.Capacity.Memory())
-		cpuAllocatable.Add(*node.Status.Allocatable.Cpu())
-		memoryAllocatable.Add(*node.Status.Allocatable.Memory())
+		for key, value := range node.Status.Capacity {
+			if _, exist := capacityList[clusterv1.ResourceName(key)]; exist {
```

Comment:
can we do here like

if cap, exist := capacityList[clusterv1.ResourceName(key)]; exist {
  cap.Add(Value)
}

Reply:
@qiujian16,
The codes were updated with your suggestions, please help review again.

Reply:
have you committed the code?

Reply:
have you committed the code?

Reply:
@qiujian16 
It was updated now. Sorry for this. :)